### ISSUE TITLE
InvalidManifest: [/opt/ros/kinetic/share/roslisp/manifest.xml] invalid XML

### ISSUE THREAD

**Issue by adrianheron** [ros-infrastructure/rospkg#154]
Full error is:
```ERROR: Rosdep experienced an error: [/opt/ros/kinetic/share/roslisp/manifest.xml] invalid XML: 'ascii' codec can't encode character u'\xf6' in position 213: ordinal not in range(128)
Please go to the rosdep page [1] and file a bug report with the stack trace below.
[1] : http://www.ros.org/wiki/rosdep

rosdep version: 0.12.2

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 140, in rosdep_main
    exit_code = _rosdep_main(args)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 388, in _rosdep_main
    return _package_args_handler(command, parser, options, args)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 484, in _package_args_handler
    return command_handlers[command](lookup, packages, options)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/main.py", line 677, in command_install
    uninstalled, errors = lookup.resolve_all(packages, installer_context, implicit=options.recursive)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/lookup.py", line 393, in resolve_all
    rosdep_keys = self.get_rosdeps(resource_name, implicit=implicit)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/lookup.py", line 315, in get_rosdeps
    return self.loader.get_rosdeps(resource_name, implicit=implicit)
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/rospkg_loader.py", line 138, in get_rosdeps
    if resource_name in self.get_catkin_paths():
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/rospkg_loader.py", line 127, in get_catkin_paths
    self._catkin_packages_cache = dict(find_catkin_paths(self._rospack))
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/rospkg_loader.py", line 126, in find_catkin_paths
    filter(lambda x: src.get_manifest(x).is_catkin, src.list()))
  File "/usr/local/lib/python2.7/dist-packages/rosdep2/rospkg_loader.py", line 126, in <lambda>
    filter(lambda x: src.get_manifest(x).is_catkin, src.list()))
  File "/usr/local/lib/python2.7/dist-packages/rospkg/rospack.py", line 167, in get_manifest
    return self._load_manifest(name)
  File "/usr/local/lib/python2.7/dist-packages/rospkg/rospack.py", line 211, in _load_manifest
    retval = self._manifests[name] = parse_manifest_file(self.get_path(name), self._manifest_name, rospack=self)
  File "/usr/local/lib/python2.7/dist-packages/rospkg/manifest.py", line 434, in parse_manifest_file
    return parse_manifest(manifest_name, f.read().decode('utf-8'), filename)
  File "/usr/local/lib/python2.7/dist-packages/rospkg/manifest.py", line 454, in parse_manifest
    raise InvalidManifest("[%s] invalid XML: %s" % (filename, e))
InvalidManifest: [/opt/ros/kinetic/share/roslisp/manifest.xml] invalid XML: 'ascii' codec can't encode character u'\xf6' in position 213: ordinal not in range(128)
```

XML in question
```<package>
  <description brief="A Lisp client library for ROS">
    This package is a Lisp client library for ROS, the Robot Operating
    System.
  </description>
  <author>Bhaskara Marthi, Brian Gerkey, Lorenz Mösenlechner, Thibault Kruse</author>
  <license>BSD</license>
  <review status="experimental" notes="" />
  <depend package="roslang" />
  <depend package="rosgraph_msgs" />
  <depend package="std_srvs" />
  <depend package="rospack" />
  <export>
    <roslang cmake="${prefix}/rosbuild/roslisp.cmake" />
  </export>
  <rosdep name="sbcl" />
  <url>http://ros.org/wiki/roslisp</url>
</package>
```

Character 213 is the umlaut in Mösenlechner

Appears to be related to https://github.com/ros-infrastructure/rospkg/pull/138, reverting those changes allows the XML to parse correctly

**Comment by HEllRZA** [comment]
Seems like manifest.py is only working with python3.

Suggested Fix: (manifest.py line 433)

```
import sys

with open(filename, 'rb') as f:       
        if (sys.version_info > (3, 0)):
                manifest_str = f.read().decode('utf-8')
        else:
                manifest_str = f.read()
        return parse_manifest(manifest_name, manifest_str, filename)
```

Maybe there is a more elegant solution, but this should work.

**Comment by dirk-thomas** [comment]
Regression of #138.

**Comment by dirk-thomas** [comment]
Addressed by #155. Will be available in version 1.1.7.
